jupyter-extension/jupyterlab_pachyderm/mount_server_client.py

```diff
@@ -15,21 +16,37 @@
 
 lock = locks.Lock()
 MOUNT_SERVER_PORT = 9002
+HTTP_UNIX_SOCKET_SCHEMA="http_unix"
+HTTP_SCHEMA="http"
 
+DEFAULT_SCHEMA=HTTP_UNIX_SOCKET_SCHEMA
+#DEFAULT_SCHEMA=HTTP_SCHEMA
 
 class MountServerClient(MountInterface):
     """Client interface for the mount-server backend."""
 
     def __init__(
         self,
         mount_dir: str,
-    ):
-        self.client = AsyncHTTPClient()
+        sock_path: str,
+        ):
+        
         self.mount_dir = mount_dir
-        self.address = f"http://localhost:{MOUNT_SERVER_PORT}"
+
+        if DEFAULT_SCHEMA == HTTP_UNIX_SOCKET_SCHEMA:
+            self.address = 'http://localhost'
+            AsyncHTTPClient.configure('tornado.curl_httpclient.CurlAsyncHTTPClient')
```

Comment:
I don't know how this object works, but should you be calling the `configure` method on the instantiated object? I have a feeling that by calling it directly on the class you are globally configuring the `AsyncHTTPClient` to have this setting.

Reply:
This looks like the correct sequence to follow otherwise it might not work. Also it is only used when we want to use the unix domain socket.